### PR TITLE
Machine-Healthcheck: Set Requeue if error in reading object

### PR DESCRIPTION
**Pull request by vikaschoudhary16**

### CODE REVIEW

File: pkg/controller/machinehealthcheck/machinehealthcheck_controller.go [hunk 1]
```diff
@@ -100,7 +100,8 @@ func (r *ReconcileMachineHealthCheck) Reconcile(request reconcile.Request) (reco
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
+		glog.Warningf("Requeuing the key. Error reading the object: %v", err)
```
Comment (enxebre) [inline pkg/controller/machinehealthcheck/machinehealthcheck_controller.go]: why is this needed? by default any requests will requeue when returning error != nil

**Comment by ingvagabund** [comment]
@vikaschoudhary16 can you update the description explaining why it is needed and what happens if not fixed? Any upstream issue or bug that you are fixing?

**Comment by vikaschoudhary16** [comment]
Agree @enxebre @ingvagabund. I was mistaken. Closing.
